app/src/main/kotlin/co/netguru/baby/monitor/client/feature/feedback/FeedbackFrequencyUseCase.kt

```diff
@@ -0,0 +1,61 @@
+package co.netguru.baby.monitor.client.feature.feedback
+
+import android.content.SharedPreferences
+import co.netguru.baby.monitor.client.application.di.FeedbackPreferencesQualifier
+import co.netguru.baby.monitor.client.common.TimestampProvider
+import co.netguru.baby.monitor.client.common.extensions.edit
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class FeedbackFrequencyUseCase @Inject constructor(
+    @FeedbackPreferencesQualifier
+    private val sharedPreferences: SharedPreferences,
+    private val timestampProvider: TimestampProvider
+) {
+    fun shouldAskForFeedback(): Boolean {
+        val currentCounter = sharedPreferences.getInt(NOTIFICATION_COUNTER_KEY, 0)
+        val lastFeedbackTimestamp = sharedPreferences.getLong(FEEDBACK_TIMESTAMP_KEY, 0)
+        return enoughTimePassed(lastFeedbackTimestamp) && currentCounter > NO_FEEDBACK_COUNTER_LIMIT
+    }
```

Comment:
I wonder if it could be modeled with rx. Something like this, perhaps?
```
fun feedbackRecordings(recordings: Observable<Recording>) =
    Observable.interval(DELAY_HOURS, TimeUnit.HOURS)
        .withLatestFrom(recordings) { _, recording -> recording }
```